src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,265 @@
+package duke;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.*;
+
+public class Duke {
+    //declarations
+    public static String line = "------------------------------------------------------------------------------------------\n";
+    private static int quitFlag = 0; //if 0, take user input. Otherwise, don't.
+
+    //declare task array and keep track of how many tasks stored
+    public static ArrayList<Task> t = new ArrayList<>();
+    public static int taskCount = 0;
+
+    //Program starts with this greeting
+    public static void start() throws FileNotFoundException, DukeException {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println(line + "Hello! I'm Duke.\n" + logo + "What can i do for you?\n" + line);
+        loadData();
+    }
+
+    //Saves Task list into local file
+    public static void saveData(ArrayList<Task> t) {
+        String path = "D:\\Documents\\NUS\\Y2S1\\CS2113T\\IP\\UserData.txt";
+        try {
+            FileWriter fw = new FileWriter(path, false);
+            PrintWriter pw = new PrintWriter(fw, false);
+            pw.flush();
+            pw.close();
+            fw.close();
+            for (int i = 0; i < taskCount; i++) {
+                String input = t.get(i).toSave() + "\n";
+                Files.write(Paths.get(path), input.getBytes(), StandardOpenOption.APPEND);
+            }
+        }  catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    //Loads Task list into Duke
+    public static void loadData() throws FileNotFoundException, DukeException {
+        File f = new File("D:\\Documents\\NUS\\Y2S1\\CS2113T\\IP\\UserData.txt");
+
+        Scanner scan = new Scanner(f);
+        String taskType;
+        String s;
+        int taskNumber = 1; //tracks how many tasks read from .txt file so far
+
+        while(scan.hasNext())                                                                                           //todo hello | 1
+        {                                                                                                               //deadline hello /by Sunday | 1
+            String data = scan.nextLine();                                                                              //event project meeting /at Mon 2-4pm | 0
```

Comment:
Perhaps you could rewrite your while loop as per https://se-education.org/guides/conventions/java/index.html#layout so that `{` appear as the same line as `while()`.